src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,41 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        int totalSalary = 0;
+        StringBuilder result = new StringBuilder();
+        LocalDate startDate = LocalDate.parse(dateFrom, FORMATTER);
+        LocalDate endDate = LocalDate.parse(dateTo, FORMATTER);
+        result.append("Report for period ").append(dateFrom).append(" - ").append(dateTo);
+        for (String name : names) {
+            for (String record : data) {
+                String[] valuesData = record.split(" ");
+                boolean isNull = false;
+                for (String value : valuesData) {
+                    if (value == null) {
+                        isNull = true;
+                        break;
+                    }
+                }
+                if (isNull) {
+                    continue;
+                }
```

Comment:
why do we need this part of code? can we remove it?

Reply:
if we have good data... no